SD2019StackOverflowApp/src/main/java/ro/utcn/sd/mca/SD2019StackOverflowApp/service/AnswerManagementService.java

```diff
@@ -0,0 +1,142 @@
+package ro.utcn.sd.mca.SD2019StackOverflowApp.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.entity.*;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.exception.InvalidAnswerIdException;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.exception.InvalidQuestionIdException;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.exception.InvalidSOUserIdException;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.persistence.RepositoryFactory;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.persistence.Specification;
+import ro.utcn.sd.mca.SD2019StackOverflowApp.persistence.SpecificationFactory;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class AnswerManagementService {
+    private final RepositoryFactory repositoryFactory;
+    private final SpecificationFactory specificationFactory;
+
+    @Transactional
+    public Answer addAnswer(Integer questionId, Integer sOUserId, String text) throws InvalidQuestionIdException, InvalidSOUserIdException {
+        Specification<Question> sq = specificationFactory.createFindQuestionById(questionId);
+        List<Question> lq = repositoryFactory.createQuestionRepository().query(sq);
+        if (lq.isEmpty())
+            throw new InvalidQuestionIdException();
+
+        Specification<SOUser> su = specificationFactory.createFindSOUserById(sOUserId);
+        List<SOUser> lu = repositoryFactory.createSOUserRepository().query(su);
+        if (lu.isEmpty())
```

Comment:
I strongly recommend that you get used to always use curly brackets (`{` and `}`) for if, else, while, for, etc. statements, even if you only have one statement inside. Check out this S.O. answer for reasoning: https://stackoverflow.com/a/8020255/7612556